core/string/string_buffer.h

```diff
@@ -147,6 +148,15 @@ String StringBuffer<SHORT_BUFFER_SIZE>::as_string() {
 	}
 }
 
+template <int SHORT_BUFFER_SIZE>
+String StringBuffer<SHORT_BUFFER_SIZE>::as_string() const {
+	if (buffer.is_empty()) {
+		return String(short_buffer, string_length);
+	} else {
+		return buffer;
```

Comment:
This needs to be `buffer.substr(0, string_length + 1)` as it might be larger than the string length

Reply:
thank you!